src/containers/deactivate-account-page/DeactivateAccountPage.tsx

```diff
@@ -0,0 +1,278 @@
+  closeConfirmation: () => void
+  onDrawerSelection: (rowNum: number) => void
+}
+
+type DeactivateAccountPageContentsProps = {
+  isMobile?: boolean
+  isLoading?: boolean
+  openConfirmation: () => void
+}
+
+type DeactivateAccountModalProps = {
+  isVisible: boolean
+  isLoading: boolean
+  onClose: () => void
+  onConfirm: () => void
+}
+
+export const DeactivateAcccountPageContents = ({
+  openConfirmation,
+  isMobile,
+  isLoading
+}: DeactivateAccountPageContentsProps) => {
+  const dispatch = useDispatch()
+  const goToSafety = useCallback(() => {
+    dispatch(pushRoute('/'))
+  }, [dispatch])
+  return (
+    <div className={cn(styles.tile, { [styles.mobile]: isMobile })}>
+      <div className={styles.header}>{messages.header}</div>
+      <ul className={styles.list}>
+        {messages.listItems.map((message, i) => (
+          <li key={i}>{message}</li>
+        ))}
+      </ul>
+      {isLoading && isMobile && <LoadingSpinnerFullPage />}
+      <div className={styles.buttons}>
+        <Button
+          className={cn(styles.button, {
+            [styles.buttonDanger]: !(isLoading && isMobile)
+          })}
+          text={messages.buttonDeactivate}
+          type={
+            isLoading && isMobile ? ButtonType.DISABLED : ButtonType.PRIMARY_ALT
+          }
+          onClick={openConfirmation}
+        />
+        <Button
+          className={styles.button}
+          text={messages.buttonSafety}
+          type={
+            isLoading && isMobile ? ButtonType.DISABLED : ButtonType.PRIMARY_ALT
+          }
+          onClick={goToSafety}
+        />
+      </div>
+    </div>
+  )
+}
+
+export const DeactivateAccountConfirmationModal = ({
+  isVisible,
+  onClose,
+  isLoading,
+  onConfirm
+}: DeactivateAccountModalProps) => {
+  return (
+    <Modal
+      bodyClassName={styles.confirmModal}
+      isOpen={isVisible}
+      onClose={onClose}
+      showDismissButton
+      showTitleHeader
+      title={messages.confirmTitle}
+    >
+      <div className={styles.container}>
+        {isLoading ? (
+          <LoadingSpinnerFullPage />
+        ) : (
+          <div className={styles.confirmText}>{messages.confirm}</div>
+        )}
+        <div className={styles.buttons}>
+          <Button
+            className={cn(styles.button, {
+              [styles.buttonDanger]: !isLoading
+            })}
+            isDisabled={isLoading}
+            onClick={onConfirm}
+            textClassName={styles.deleteButtonText}
+            text={messages.buttonDeactivate}
+            type={isLoading ? ButtonType.DISABLED : ButtonType.PRIMARY_ALT}
+          />
+          <Button
+            className={styles.button}
+            isDisabled={isLoading}
+            onClick={onClose}
+            text={messages.buttonGoBack}
+            type={isLoading ? ButtonType.DISABLED : ButtonType.PRIMARY_ALT}
+          />
+        </div>
+      </div>
+    </Modal>
+  )
+}
+
+export const DeactivateAccountPageDesktop = ({
```

Comment:
let's pull these into 3 components

DeactivateAccountPage.tsx
mobile/DeactivateAccount.tsx
desktop/DeactivateAccount.tsx

and perhaps even follow the DeactivateAccountProvider.tsx model we have set up for something like the profile page.

Reply:
done!